src/scss/tools/functions/_units.scss

```diff
@@ -68,6 +51,23 @@ $line-height-base : line-height(); // Lifted out of the variables file to fix ci
     }
 }
 
+@function line-height($font-size: 'body-s', $line-height: '20', $scale: 'default') {
+    @if type-of($font-size) == 'number' {
+        @return decimal-round(math.div($line-height, $font-size), 2);
+    } @else if map-has-key(v.$type, $font-size) { // else try and find the value in our type map
+        $key-map: map-get(v.$type, $font-size);
+        $font-list: map-get($key-map, $scale);
+
+        @if type-of($font-list) == 'list' {
+            @return line-height(nth($font-list, 1), nth($font-list, 2));
+        } @else {
+            @warn 'No line-height found as part of the $type map for #{ $font-size }';
+        }
+    }
+}
+
+$line-height-base : line-height(); // Lifted out of the variables file to fix circular reference issues
```

Comment:
Out of interest, is there a reason this needs to move down below the functions? Feels a bit hidden away

Reply:
Ah ok – so it's a hoisting issue essentially. No worries was just wondering!